### PR TITLE
Public promotion

### PR DESCRIPTION
**Pull request by gary1998**
- updated examples
- updated website docs
- added occurrence api support
- updated scc-go-sdk

### CODE REVIEW

**Review by reeteshsingh93 — APPROVED** [review]

File: ibm/resource_ibm_scc_si_occurrence.go [hunk 1]
```diff
@@ -0,0 +1,844 @@
...
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The unique URL of the resource, image or the container, for which the `Occurrence` applies. For example, https://gcr.io/provider/image@sha256:foo. This field can be used as a filter in list requests.",
+			},
+			"remediation": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "A description of actions that can be taken to remedy the `Note`.",
+			},
+			"create_time": &schema.Schema{
```
Comment (hkantare) [inline ibm/resource_ibm_scc_si_occurrence.go:66]: create_time  and update_time is a computed why are you reading that attribute in create /update

File: website/docs/d/scc_si_occurrence.html.markdown [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+* `account_id` - (Optional, String) Account ID is optional, if not provided value will be inferred from the token retrieved from the IBM Cloud API key.
+* `occurrence_id` - (Required, Forces new resource, String) Second part of occurrence `name`: providers/{provider_id}/occurrences/{occurrence_id}.
+* `provider_id` - (Required, Forces new resource, String) Part of the parent. This field contains the provider ID. For example: providers/{provider_id}.
+
+## Attribute Reference
+
+In addition to all argument references listed, you can access the following attribute references after your data source is created.
+
+* `id` - The unique identifier of the scc_si_occurence.
+* `context` - (Optional, List)
```
Comment (hkantare) [inline website/docs/d/scc_si_occurrence.html.markdown:35]: In attribute Refernce why the attributes have optional ?

File: ibm/data_source_ibm_scc_si_occurrences.go [hunk 3]
```diff
@@ -0,0 +1,586 @@
...
+			"account_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"provider_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Part of the parent. This field contains the provider ID. For example: providers/{provider_id}.",
+			},
+			"page_size": &schema.Schema{
```
Comment (hkantare) [inline ibm/data_source_ibm_scc_si_occurrences.go:32]: For all ibm_scc_si datasources which gives complete list can we page_size and page_token and use the pagination to get complete list something similar to
data_source_ibm_scc_posture_profiles.go datasource

Comment (gary1998) [reply ibm/data_source_ibm_scc_si_occurrences.go:32]: there are 3 data sources that give a list in response and all of them already have pagination
- `ibm_scc_si_providers` asks `limit` and `skip`
- `ibm_scc_si_notes` asks `page_size` and `page_token`
- `ibm_scc_si_occurrences` asks `page_size` and `page_token`

and if you're asking about `first`, `last` and `previous` options, those are not supported by our backend APIs as for now.

File: ibm/resource_ibm_scc_si_occurrence.go (hunk 1 above)
Comment (gary1998) [reply ibm/resource_ibm_scc_si_occurrence.go:66]: removed

File: ibm/data_source_ibm_scc_si_occurrences.go (hunk 3 above)
Comment (gary1998) [reply ibm/data_source_ibm_scc_si_occurrences.go:32]: as discussed, added the code for getting complete result for `ListNotes` and `ListProviders` in data sources.

**Pull request closed without merging**